packages/legend-extension-dsl-data-space/src/graph-manager/protocol/pure/v1/V1_DSL_DataSpace_PureGraphManagerExtension.ts

```diff
@@ -211,6 +215,70 @@ export class V1_DSL_DataSpace_PureGraphManagerExtension extends DSL_DataSpace_Pu
     );
   }
 
+  async analyzeDataSpaceCoverage(
+    dataSpacePath: string,
+    entitiesRetriever: () => Promise<Entity[]>,
+    cacheRetriever?: () => Promise<PlainObject<DataSpaceAnalysisResult>>,
+    actionState?: ActionState,
+    graphReport?: GraphManagerOperationReport | undefined,
+    pureGraph?: PureModel | undefined,
+    executionContext?: string | undefined,
+    mappingPath?: string | undefined,
+    projectInfo?: ProjectGAVCoordinates,
+    notificationService?: NotificationService | undefined,
+  ): Promise<DataSpaceAnalysisResult> {
+    const cacheResult = cacheRetriever
+      ? await this.fetchDataSpaceAnalysisFromCache(cacheRetriever, actionState)
+      : undefined;
+    const engineClient = this.graphManager.engine.getEngineServerClient();
+    let analysisResult: PlainObject<V1_DataSpaceAnalysisResult>;
+    if (
+      cacheResult &&
+      V1_deserializeDataSpaceAnalysisResult(
+        cacheResult,
+        this.graphManager.pluginManager.getPureProtocolProcessorPlugins(),
+      ).executionContexts.every(
+        (e) => e.mappingModelCoverageAnalysisResult.model !== undefined,
+      )
+    ) {
+      analysisResult = cacheResult;
+    } else {
+      notificationService?.notify(
```

Comment:
HEADS-UP: I moved this notification inside the else block from where it was originally added above the if statement